.github/workflows/tf-azure-deploy-dev.yml

```diff
@@ -0,0 +1,133 @@
+name: 'Terraform Deploy [Azure - DEV]'
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - 'terraform-azure/**'
+      - '!terraform-azure/**.md'
+
+defaults:
+  run:
+    working-directory: ./terraform-azure
+
+# Permissions for OIDC authentication
+permissions:
+  id-token: write
+  contents: read
+  pull-requests: write
+
+# Used by the Terraform Azure Provider for OIDC
+env:
+  ARM_CLIENT_ID: "${{ secrets.AZURE_CLIENT_ID }}"
```

Comment:
The secrets `AZURE_CLIENT_ID` and `AZURE_SUBSCRIPTION_ID` are only set in the `development` environment. Does that mean you want:

```yaml
jobs:
  terraform-plan:
    name: Terraform Plan
    runs-on: ubuntu-latest
    environment: development
    env:
      AZURE_CLIENT_ID: foo
      AZURE_SUBSCRIPTION_ID: bar
```
https://github.com/DFE-Digital/early-years-foundation-recovery/settings/environments/435279666/edit

Reply:
Yes it does, good spot and thanks!

Reply:
I am instead going to set placeholders for Test and Production as these variables play a role in OIDC authentication.